### PR TITLE
New PowerShell Cheat Sheet

### PR DESCRIPTION
**Pull request by AnthonyNeace**
This is a cheat sheet for syntax and operators in the PowerShell language.  Most examples are from the language's builtin docs that can also be found [here](https://technet.microsoft.com/en-us/library/hh847856.aspx).

IA: https://duck.co/ia/view/powershell_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@AnthonyNeace Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by gautamkrishnar** [comment]
Hey @AnthonyNeace Welcome to duckduckhack :smile:

File: share/goodie/cheat_sheets/json/powershell.json [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+            {
+                "key": "\"MyString\" -like \"*String\"",
+                "val": "Match using the wildcard character (*)"
+            },        
+            {
+                "key": "\"MyString\" -notlike \"Other*\"",
+                "val": "Does not match using the wildcard character (*)"
+            },    
+            {
+                "key": "\"MyString\" -match \"String\"",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/powershell.json]: Can you change the key to:
`"key": "\"MyString\" -match \"^String$\""` so it's more obvious that it's regex?

File: share/goodie/cheat_sheets/json/powershell.json [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+                "val": "Matches a string using regular expressions"
+            },    
+            {
+                "key": "\"MyString\" -notmatch \"Other\"",
+                "val": "Does not match a string using regular expressions"
+            },         
+            {
+                "key": "\"abc\", \"def\" -contains \"def\"",
+                "val": "Returns boolean true only when the test value (right) exactly matches at least one of the reference values (left)"
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/powershell.json]: Hmm, I wonder if we can rephrase this as "Returns true if the right argument is in the present in the array (left)" or similar?

File: share/goodie/cheat_sheets/json/powershell.json [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+            {
+                "key": "$( exp1; exp2 )",
+                "val": "Subexpression operator; Returns the result of one or more contained statements"
+            },    
+            {
+                "key": "@( exp1; exp2 )",
+                "val": "Array subexpression operator; Returns the result of one or more contained statements as an array"
+            },       
+            {
+                "key": "\\[datetime\\]::now",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/powershell.json]: Can we change this to `DateTime`; I appreciate that Powershell is case insensitive however I think it's better if the names are accurate to the .NET framework

File: share/goodie/cheat_sheets/json/powershell.json [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+                "val": "Static member operator; Calls the static properties operator and methods of a .NET Framework class"
+            },       
+            {
+                "key": "$arr = 1,2,3",
+                "val": "Comma operator (binary); the comma creates an array"
+            },     
+            {
+                "key": "$arr = ,1",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/powershell.json]: Alternative to this is `$arr = @(1)` which looks a little cleaner, however I think we should keep the floating `,` in here too as people do use it

**Comment by AnthonyNeace** [comment]
@gautamkrishnar Thanks!

@mintsoft Thanks for the feedback, implemented it in the latest commit.

File: share/goodie/cheat_sheets/json/powershell.json [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+    "metadata": {
+        "sourceName": "TechNet",
+        "sourceUrl" : "https://technet.microsoft.com/en-us/library/hh847856.aspx"
+    },
+
+    "aliases": [
+        "power shell", "posh"
+    ],
+
+    "template_type": "code",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/powershell.json]: @AnthonyNeace let's use the `terminal` template -- this will look much better!

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=powershell cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/powershell_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@AnthonyNeace Thanks! This LGTM :+1:
